components/Main/index.js

```diff
@@ -1,22 +1,155 @@
+//  AA!243232
+const threeDotsIcon = document.querySelector(".three-dots-icon");
+const redDeleteIcon = document.querySelector(".red-delete-icon");
+const greenTrueIcon = document.querySelector(".green-true-icon");
+let wantedProject;
 export const viewProjectNameOnMainPage = () => {
-  let wantedProject = getProjectFromLocalStorageById(
-    +localStorage.getItem("currentProjectIdFromSidebar")
+  wantedProject = getProjectById(
+    getLocalStorage("currentProjectIdFromSidebar")
   );
+  if (!wantedProject) {
+    proName.textContent = "NoProjectsYet";
+    threeDotsIcon.classList.add("hide");
+  } else {
+    proName.textContent = wantedProject.projectName;
+    threeDotsIcon.classList.remove("hide");
+  }
+};
+const actionMenuContainer = document.querySelector(".actions-container");
+export const deleteProjectUiAction = (deletedId) => {
+  let projects = getLocalStorage("projects");
+  const project = projects.find((item) => item.ProjectId == deletedId);
+  let currantIndex = projects.indexOf(project);
+
+  if (currantIndex == 0 && projects.length > 1) {
+  } else {
+    currantIndex--;
+  }
+
+  projects = deleteProject(deletedId);
+
+  const newId = currantIndex < 0 ? null : projects[currantIndex].ProjectId;
+
+  renderCurrentProject(newId);
+  previewProjectsNameOnTheSidebar();
+
+  const liElement = document.getElementById(newId);
+
+  if (liElement) {
+    liElement.classList.add("selected-project");
+  }
+};
+
+let actionForm = document.forms["actionForm"]["edit-project-name-input"];
+
+const editProjectUiAction = () => {
+  //copied code
+  let EditProjectNameInput = actionForm;
+  wantedProject = getProjectById(
+    getLocalStorage("currentProjectIdFromSidebar")
+  );
+  EditProjectNameInput.value = wantedProject.projectName;
+  function isValidString(input) {
+    const regex = /^[a-zA-Z0-9 ]*$/;
+    return regex.test(input);
+  }
+  let errorMSG = document.createElement("div");
+
+  errorMSG.style.color = "red";
+  errorMSG.style.fontSize = "11px";
+  errorMSG.style.fontWeight = "500";
+  errorMSG.style.position = "absolute";
+  errorMSG.style.left = " 305px";
+  errorMSG.style.top = " 80px";
 
-  proName.textContent = wantedProject.projectName;
+  const lableForProjectNameInput = document.querySelector(
+    ".lable-for-project-name-input"
+  );
+  EditProjectNameInput.onblur = function () {
+    if (!isValidString(EditProjectNameInput.value)) {
+      errorMSG.textContent = "*Please Enter a Valid Name Only 'a-z A-Z 0-9'";
+
+      EditProjectNameInput.style.border = "1px red solid";
+      EditProjectNameInput.before(errorMSG);
+    }
+
+    if (EditProjectNameInput.value == "") {
+      errorMSG.textContent = "*Project Name Can Not Be Empty'";
+
+      EditProjectNameInput.before(errorMSG);
+      EditProjectNameInput.style.border = "1px red solid";
+    }
+  };
+
+  EditProjectNameInput.onfocus = function () {
+    EditProjectNameInput.style.border = "";
+
+    errorMSG.remove();
+  };
```

Comment:
1. Extract the error display logic into a reusable function to avoid code duplication.
2. Use `addError` and `removeError` utility functions to manage the styling and message.

```js
function addError(inputElement, message) {
    errorMsg.textContent = message;
    inputElement.style.border = "1px solid red";
    inputElement.before(errorMsg);
}

function removeError(inputElement) {
    inputElement.style.border = "";
    errorMsg.remove();
}

EditProjectNameInput.onblur = function () {
    if (!isValidString(EditProjectNameInput.value)) {
        addError(EditProjectNameInput, "*Please Enter a Valid Name Only 'a-z A-Z 0-9'");
    } else if (EditProjectNameInput.value === "") {
        addError(EditProjectNameInput, "*Project Name Cannot Be Empty");
    }
};

EditProjectNameInput.onfocus = function () {
    removeError(EditProjectNameInput);
};
```